cmd/draft/connect.go

```diff
@@ -96,41 +105,85 @@ func (cn *connectCmd) run(runningEnvironment string) (err error) {
 
 	var connectionMessage = "Your connection is still active.\n"
 
+	exportEnv := make(map[string]string)
+
 	// output all local ports first - easier to spot
 	for _, cc := range connection.ContainerConnections {
 		for _, t := range cc.Tunnels {
-			err = t.ForwardPort()
-			if err != nil {
+			if err = t.ForwardPort(); err != nil {
 				return err
 			}
-			m := fmt.Sprintf("Connect to %v:%v on localhost:%#v\n", cc.ContainerName, t.Remote, t.Local)
-
-			connectionMessage += m
-			fmt.Fprintf(cn.out, m)
-		}
-	}
-
-	for _, cc := range connection.ContainerConnections {
-		readCloser, err := connection.RequestLogStream(deployedApp.Namespace, cc.ContainerName, cn.logLines)
-		if err != nil {
-			return err
+			if export {
+				var (
+					application = strings.Replace(strings.ToUpper(deployedApp.Name), "-", "_", -1)
+					container   = strings.ToUpper(cc.ContainerName)
+					prefix      = fmt.Sprintf("%s_%s", application, container)
+				)
+				exportEnv[prefix+"_SERVICE_HOST"] = fmt.Sprintf("localhost")
+				exportEnv[prefix+"_SERVICE_PORT"] = fmt.Sprintf("%#v", t.Local)
+			} else {
+				m := fmt.Sprintf("Connect to %v:%v on localhost:%#v\n", cc.ContainerName, t.Remote, t.Local)
+				connectionMessage += m
+				fmt.Fprintf(cn.out, m)
+			}
 		}
-
-		defer readCloser.Close()
-		go writeContainerLogs(cn.out, readCloser, cc.ContainerName)
 	}
 
-	stop := make(chan os.Signal, 2)
+	stop := make(chan os.Signal, 1)
+	done := make(chan struct{})
 	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-stop
-		os.Exit(0)
+		close(done)
 	}()
 
-	for {
-		time.Sleep(5 * time.Minute)
-		fmt.Fprintf(cn.out, connectionMessage)
+	if export {
+		for env, val := range exportEnv {
+			fmt.Fprintf(os.Stdout, "export %s=%s\n", env, val)
+		}
+		os.Stdout.Close()
+		<-done
+		return nil
+	} else {
+		for _, cc := range connection.ContainerConnections {
+			readCloser, err := connection.RequestLogStream(deployedApp.Namespace, cc.ContainerName, cn.logLines)
+			if err != nil {
+				return err
+			}
+			defer readCloser.Close()
+			go writeContainerLogs(cn.out, readCloser, cc.ContainerName)
+		}
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				fmt.Fprintf(cn.out, connectionMessage)
+			case <-done:
+				return nil
+			}
+		}
+	}
+}
+
+func (cn *connectCmd) detach() error {
+	cmd := exec.Command("draft", "connect", "--export")
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	cmd.Env = os.Environ()
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+	b, err := ioutil.ReadAll(stdout)
+	if err != nil {
+		return err
 	}
+	cmd.Process.Release()
+	fmt.Println(string(b))
```

Comment:
since stdout from `draft connect --export` already prints an extra newline, can we just call `fmt.Print()` instead, or trim whitespace via `fmt.Println(strings.TrimSpace(string(b)))`? The output right now looks like this:

```
><> draft connect --detach
export EXAMPLE_GO_GO_SERVICE_HOST=localhost
export EXAMPLE_GO_GO_SERVICE_PORT=35505

><>
```

Reply:
Also, what do you think about writing a few comments at the end of the string to make it a bit more user-friendly? e.g. something like this:

```
fmt.Printf("%s\n# # Run this command to configure your shell:\n# eval $(%s connect --detach)\n\n# To kill the underlying process:\n# sudo kill %d\n", strings.TrimSpace(string(b)), os.Args[0], cmd.Process.Pid)
```

Reply:
revised edit:

```
fmt.Printf("%s\n# Run this command to configure your shell:\n# eval $(%s connect --detach)\n", strings.TrimSpace(string(b)), os.Args[0])
```

Reply:
and for Windows, something equivalent to https://github.com/fishworks/gofish/blob/master/cmd/gofish/tank_windows.go#L21 would be +:100:

Reply:
Given how this works, It wouldn't make much sense to instruct them to run the eval command on export because re-running will spawn a new detached draft process. It would be nice to be able to re-export the same environment on every detach if one already exists but there's no way to do this just yet because detached draft's are not tracked between invocations.